File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
    let maxi = 0;
    let ex = prices[0];
    
    for (let p of prices) {
        if (p > ex) maxi += p - ex;
        ex = p;
    }
    
    return maxi;
};

File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxi = 0;
        int bf = prices[0];

        for (size_t i = 0; i < prices.size(); i++) {
            int price = prices[i];
            if (price > bf) maxi += price - bf;
            bf = price;
        }
        
        return maxi;
    }
};